src/AI/CommanderProductionInfo.java

```diff
@@ -15,56 +16,59 @@
 import Units.UnitModel;
 
 /**
- * Keeps track of a commander's production facilities. When created, it will automatically catalog
+ * Keeps track of a army's production facilities. When created, it will automatically catalog
  * all available facilities, and all units that can be built. It is then easy to ask whether it is
  * possible to build a given type of unit, or find a location to do so.
  * Once a purchase has been scheduled, removeBuildLocation() will remove a given facility from any
  * further consideration.
  */
 public class CommanderProductionInfo
 {
-  public Commander myCo;
-  public Set<UnitModel> availableUnitModels;
+  public Army myArmy;
+  public Set<ModelForCO> availableUnitModels;
   public Set<MapLocation> availableProperties;
-  public Map<Terrain.TerrainType, Integer> propertyCounts;
+  public Map<Commander, Map<Terrain.TerrainType, Integer>> propertyCounts;
   public Map<UnitModel, Set<TerrainType>> modelToTerrainMap;
 
   /**
    * Build a model of the production capabilities for a given Commander.
```

Comment:
Commander -> Army, I suppose.

Reply:
No, actually. Commanders can have different prices for the same unit.

Reply:
OK, fair, but the function is taking in an Army, not a Commander, and the prices for the latter can be taken abstractly under the umbrella of the former.
Modeling the production capabilities of an Army inherently requires modeling the underlying Commanders, which (I believe) this does now.

Reply:
Oh, you're talking about the comment that didn't get updated.
Fair enough.